src/libcore/result.rs

```diff
@@ -501,6 +501,8 @@ impl<T, E> Result<T, E> {
 
     /// Returns an iterator over the possibly contained value.
     ///
+    /// The iterator yields one value if the result is [`Ok`], otherwise none.
```

Comment:
It's been discussed for a while. You can take a look at [this issue](https://github.com/rust-lang/rust/issues/38400) if you want to take a look at the current status.